File: src/main/java/workshop/common/fraudwordsservice/FraudWordService.java
package workshop.common.fraudwordsservice;

import javaslang.collection.List;

public class FraudWordService {

    // returns a list of banned words from any list of words
    public List<FraudWord> examineWords(List<String> words) {
        return words
                .map(String::trim)
                .filter(s -> !s.isEmpty())
                .filter(w -> allWords().map(FraudWord::getWord).contains(w))
                .map(FraudWord::new);
    }

    private List<FraudWord> allWords() {
        return List.of("westernunion", "advance", "nigeria", "wiretransfer").map(FraudWord::new);
    }
}

File: src/main/java/workshop/common/fraudwordsservice/FraudWordService.kt
package workshop.common.fraudwordsservice

class FraudWordService {

    fun examineWords(words: List<String>): List<FraudWord> {
        return words
                .map { it.trim() }
                .filter { !it.isEmpty() }
                .filter { allWords().contains(FraudWord(it)) }
                .map{ FraudWord(it) }
    }

    private fun allWords(): List<FraudWord> {
        return listOf("westernunion", "advance", "nigeria", "wiretransfer").map{ FraudWord(it) }
    }
}
